reconcile/utils/saasherder/models.py

```diff
@@ -81,6 +94,18 @@ def state_key(self) -> str:
             key += f"/{self.target_name}"
         return key
 
+    def extract_slo_keys(self) -> list[SLOKey]:
+        return [
+            SLOKey(
+                slo_document_name=slo_document.name,
+                namespace_name=namespace.namespace.name,
+                cluster_name=namespace.namespace.cluster.name,
+            )
+            for slo_document in self.slos or []
+            for namespace in slo_document.namespaces
+            if namespace.namespace.name == self.namespace_name
```

Comment:
may need to filter by cluster name as well, as namespace name can be same cross multiple clusters